infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -102,14 +102,16 @@ public class LogReplicationSinkManager implements DataReceiver {
 
     private ISnapshotSyncPlugin snapshotSyncPlugin;
 
-    private ExecutorService applyExecutor;
+    // TODO V2: tune the thread count
+    private static ExecutorService applyExecutor = Executors.newFixedThreadPool(2,
+            new ThreadFactoryBuilder()
+                    .setDaemon(true)
+                    .setNameFormat("snapshotSyncApplyExecutor-%d")
+                    .build());;
```

Comment:
that would entail pulling the initialization of this thread pool to a new class. I am not fully convinced to do that, since there will be no other members in that class except for the thread pool initialization.